packages/rollup-core/src/app/data/producers/l2-chain-data-persister.ts

```diff
@@ -140,6 +144,23 @@ export class L2ChainDataPersister extends ChainDataProcessor {
       )}${response.v.toString(16)}`
     }
 
+    log.debug(
+      `L2 Tx Output for block ${receipt.blockNumber}: ${JSON.stringify(res)}`
+    )
+
     return res
   }
+
+  private static parseBigNumber(data: any): BigNumber {
+    if (!data) {
+      return undefined
+    }
+    if (typeof data === 'string') {
+      return new BigNumber(remove0x(data))
```

Comment:
```suggestion
      return new BigNumber(remove0x(data), 16)
```
Not positive this is the correct syntax, but have previously ran into a gnarly bug where not specifying the radix results in hex strings which don't happen to contain `a-f` getting interpreted as decimals.

Reply:
The default encoding is 'hex' for BigNumber, but yeah, I can specify it to be clear